cmd/kubectl-gadget/biolatency.go

```diff
@@ -0,0 +1,145 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+)
+
+const (
+	GADGET_NAME        = "biolatency"
+	GADGET_OUTPUTMODE  = "Status"
+	GADGET_OUTPUTSTATE = "Completed"
+)
+
+var biolatencyTraceConfig = &utils.TraceConfig{
+	GadgetName:       GADGET_NAME,
+	TraceOutputMode:  GADGET_OUTPUTMODE,
+	TraceOutputState: GADGET_OUTPUTSTATE,
+	CommonFlags:      &params,
+}
+
+var biolatencyCmd = &cobra.Command{
+	Use:   "biolatency",
+	Short: "Generate a histogram with the distribution of block device I/O latency",
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		// Ensure parent's PersistentPreRunE is still called
+		if cmd.Parent() != nil && cmd.Parent().PersistentPreRunE != nil {
+			err := cmd.Parent().PersistentPreRunE(cmd.Parent(), args)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Biolatency does not support filtering so we need to avoid adding
+		// the default namespace configured in the kubeconfig file.
+		if params.Namespace != "" && !params.NamespaceOverridden {
+			params.Namespace = ""
+		}
+
+		return nil
+	},
+}
+
+var biolatencyStartCmd = &cobra.Command{
+	Use:          "start",
+	Short:        "Start monitor the block device I/O (disk I/O) and records the distribution of I/O latency (time)",
+	RunE:         runBiolatencyStart,
+	SilenceUsage: true,
+}
+
+var biolatencyStopCmd = &cobra.Command{
+	Use:          "stop <trace-id>",
+	Short:        "Stop monitoring and generate a report (a histogram graph) with the distribution of block device I/O latency",
+	RunE:         runBiolatencyStop,
+	SilenceUsage: true,
+}
+
+var biolatencyListCmd = &cobra.Command{
+	Use:          "list",
+	Short:        "List the currently running biolatency traces",
+	RunE:         runBiolatencyList,
+	SilenceUsage: true,
+}
+
+func init() {
+	biolatencyCmd.AddCommand(biolatencyStartCmd)
+	biolatencyCmd.AddCommand(biolatencyStopCmd)
+	biolatencyCmd.AddCommand(biolatencyListCmd)
+
+	utils.AddCommonFlags(biolatencyCmd, &params)
+	rootCmd.AddCommand(biolatencyCmd)
+}
+
+func runBiolatencyStart(cmd *cobra.Command, args []string) error {
+	if params.Node == "" {
+		return fmt.Errorf("%s only works with --node", GADGET_NAME)
+	}
+
+	biolatencyTraceConfig.Operation = "start"
+	traceID, err := utils.CreateTrace(biolatencyTraceConfig)
+	if err != nil {
+		return fmt.Errorf("failure creating trace: %v", err)
```

Comment:
%v -> %w (applies to other places as well)

Reply:
Sorry for falling into this mistake again. I changed the way I was printing the error and I forgot to update this.

Reply:
We should create a bot that comments this mistake automatically. :laughing: